src/toil/lib/aws/iam.py

```diff
@@ -20,42 +24,102 @@
-                                  "ec2:CreateTags",
-                                  "ec2:DeleteSecurityGroup",
-                                  "ec2:DescribeAvailabilityZones",
-                                  "ec2:DescribeImages",
-                                  "ec2:DescribeInstances",
-                                  "ec2:DescribeInstanceStatus",
-                                  "ec2:DescribeKeyPairs",
-                                  "ec2:DescribeSecurityGroups",
-                                  "ec2:DescribeSpotInstanceRequests",
-                                  "ec2:DescribeSpotPriceHistory",
-                                  "ec2:DescribeVolumes",
-                                  "ec2:ModifyInstanceAttribute",
-                                  "ec2:RequestSpotInstances",
-                                  "ec2:RunInstances",
-                                  "ec2:StartInstances",
-                                  "ec2:StopInstances",
-                                  "ec2:TerminateInstances",
-                                  ]
+CLUSTER_LAUNCHING_PERMISSIONS = [
+    "iam:CreateRole",
+    "iam:CreateInstanceProfile",
+    "iam:TagInstanceProfile",
+    "iam:DeleteRole",
+    "iam:DeleteRoleProfile",
+    "iam:ListAttatchedRolePolicies",
+    "iam:ListPolicies",
+    "iam:ListRoleTags",
+    "iam:PassRole",
+    "iam:PutRolePolicy",
+    "iam:RemoveRoleFromInstanceProfile",
+    "iam:TagRole",
+    "ec2:AuthorizeSecurityGroupIngress",
+    "ec2:CancelSpotInstanceRequests",
+    "ec2:CreateSecurityGroup",
+    "ec2:CreateTags",
+    "ec2:DeleteSecurityGroup",
+    "ec2:DescribeAvailabilityZones",
+    "ec2:DescribeImages",
+    "ec2:DescribeInstances",
+    "ec2:DescribeInstanceStatus",
+    "ec2:DescribeKeyPairs",
+    "ec2:DescribeSecurityGroups",
+    "ec2:DescribeSpotInstanceRequests",
+    "ec2:DescribeSpotPriceHistory",
+    "ec2:DescribeVolumes",
+    "ec2:ModifyInstanceAttribute",
+    "ec2:RequestSpotInstances",
+    "ec2:RunInstances",
+    "ec2:StartInstances",
+    "ec2:StopInstances",
+    "ec2:TerminateInstances",
+]
 
 AllowedActionCollection = Dict[str, Dict[str, List[str]]]
 
+
+@retry(errors=[AWSServerErrors])
+def delete_iam_role(role_name: str, region: Optional[str] = None, quiet: bool = True) -> None:
+    # TODO: the Boto3 type hints are a bit oversealous here; they want hundreds
+    # of overloads of the client-getting methods to exist based on the literal
+    # string passed in, to return exactly the right kind of client or resource.
+    # So we end up having to wrap all the calls in casts, which kind of defeats
+    # the point of a nice fluent method you can call with the name of the thing
+    # you want; we should have been calling iam_client() and so on all along if
+    # we wanted MyPy to be able to understand us. So at some point we should
+    # consider revising our API here to be less annoying to explain to the type
+    # checker.
```

Comment:
This comment doesn't make any sense here. But what we should have here is a docstring.

It looks like this is just moving, but we could take the opportunity to fix it.